### PR TITLE
Restore correct function of nested `HasMany` and `MorphOne` mutations

### PR DESCRIPTION
**Pull request by Carlwirkus**
Here is a draft PR with a test showing the regression from `v6.37.1`. 

The HasOne test shows that using `update` does not associate the task with the user.

The MorphOne test shows that using `update` does not associate the image with the task.

Resolves https://github.com/nuwave/lighthouse/issues/2590

### CODE REVIEW

File: tests/Integration/Execution/MutationExecutor/MorphOneTest.php [hunk 1]
```diff
@@ -77,6 +106,57 @@ final class MorphOneTest extends DBTestCase
     }
     ' . self::PLACEHOLDER_QUERY;
 
+    public function testDeeplyNestedMorphOne(): void
+    {
```
Comment (Carlwirkus) [inline tests/Integration/Execution/MutationExecutor/MorphOneTest.php:110]: This test passes. I think the issue is related to updating existing models.

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php [hunk 2]
```diff
@@ -834,4 +834,54 @@ public function testConnectModelWithCustomKey(): void
             ],
         ]);
     }
+
+
```
Comment (spawnia) [inline tests/Integration/Execution/MutationExecutor/HasManyTest.php:838]:
```suggestion

```

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php [hunk 3]
```diff
@@ -834,4 +834,54 @@ public function testConnectModelWithCustomKey(): void
...
+    public function testUpdateNestedHasMany(): void
+    {
+        $user = factory(User::class)->create();
+        $user->save();
+        assert($user instanceof User);
```
Comment (spawnia) [inline tests/Integration/Execution/MutationExecutor/HasManyTest.php:843]:
```suggestion
        assert($user instanceof User);
```

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php [hunk 4]
```diff
@@ -834,4 +834,54 @@ public function testConnectModelWithCustomKey(): void
...
+
+        $task = factory(Task::class)->create();
+        assert($task instanceof Task);
+
+        $this->graphQL(/** @lang GraphQL */ '
+            mutation UpdateUser($input: UpdateUserInput!){
```
Comment (spawnia) [inline tests/Integration/Execution/MutationExecutor/HasManyTest.php:849]:
```suggestion
            mutation ($input: UpdateUserInput!) {
```

File: tests/Integration/Execution/MutationExecutor/MorphOneTest.php [hunk 5]
```diff
@@ -369,4 +369,46 @@ public function testUpdateAndDeleteMorphOne(string $action): void
...
+    public function testNestedConnectMorphOne(): void
+    {
+        $task = factory(Task::class)->create();
+        assert($task instanceof Task);
+
+        $image = factory(Image::class)->create();
+        assert($image instanceof Image);
+
+        $this->graphQL(/** @lang GraphQL */ '
+        mutation UpdateTask($input: UpdateTaskInput!){
```
Comment (spawnia) [inline tests/Integration/Execution/MutationExecutor/MorphOneTest.php:382]:
```suggestion
        mutation ($input: UpdateTaskInput!) {
```

**Review by spawnia — CHANGES_REQUESTED** [review]

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php (hunk 2 above)
Comment (Carlwirkus) [reply tests/Integration/Execution/MutationExecutor/HasManyTest.php]: done

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php (hunk 3 above)
Comment (Carlwirkus) [reply tests/Integration/Execution/MutationExecutor/HasManyTest.php]: done

File: tests/Integration/Execution/MutationExecutor/MorphOneTest.php (hunk 5 above)
Comment (Carlwirkus) [reply tests/Integration/Execution/MutationExecutor/MorphOneTest.php]: done

File: tests/Integration/Execution/MutationExecutor/HasManyTest.php (hunk 4 above)
Comment (Carlwirkus) [reply tests/Integration/Execution/MutationExecutor/HasManyTest.php]: done

**Pull request merged**